api/app/routers/hfi_calc.py

```diff
@@ -0,0 +1,86 @@
+# pylint: disable=too-many-locals
+async def get_fire_centres(response: Response):
+    """ Returns list of fire centres and planning area for each fire centre,
+    and weather stations within each planning area. Also returns the assigned fuel type
+    for each weather station. """
+    try:
+        logger.info('/hfi-calc/fire-centres')
+        response.headers["Cache-Control"] = "max-age=0"  # don't let the browser cache this
+
+        with app.db.database.get_read_session_scope() as session:
+            # Fetch all fire weather stations from the database.
+            station_query = get_fire_weather_stations(session)
+            # Prepare a dictionary for storing station info in.
+            station_info_dict = {}
+            # Prepare empty data structures to be used in HFIWeatherStationsResponse
+            fire_centres_list = []
+            planning_areas_by_fire_center = {}
+            stations_by_planning_area = {}
+            # Iterate through all the database records, collecting all the data we need.
+            for (station_record, fuel_type_record, planning_area_record, fire_centre_record) in station_query:
+                fire_centre = FireCentre(name=fire_centre_record.name)
+                planning_area = PlanningArea(name=planning_area_record.name, fire_centre=fire_centre)
+                fire_centres_list.append(fire_centre)
+                planning_areas_by_fire_center.get(fire_centre.name).append(planning_area)
+                station_info_dict[station_record.station_code] = {
+                    'fuel_type': FuelType(
+                        abbrev=fuel_type_record.abbrev,
+                        description=fuel_type_record.description),
+                    'planning_area': planning_area
+                }
+
+            # We're still missing some data that we need from wfwx, so give it the list of stations
+            wfwx_stations_data = await wildfire_one.get_stations_by_codes(list(station_info_dict.keys()))
+            # Iterate through all the stations from wildfire one.
+            for wfwx_station in wfwx_stations_data:
+                station_info = station_info_dict[wfwx_station.code]
+                # Combine everything.
+                station_properties = WeatherStationProperties(
+                    name=wfwx_station.name,
+                    fuel_type=station_info['fuel_type'],
+                    elevation=wfwx_station.elevation,
+                    wfwx_station_uuid=wfwx_station.wfwx_station_uuid)
+
+                weather_station = WeatherStation(code=wfwx_station.code,
+                                                 station_props=station_properties,
+                                                 planning_area=station_info['planning_area'])
+
+                stations_by_planning_area.get(station_info['planning_area']).append(weather_station)
+
+            return HFIWeatherStationsResponse(fire_centres=fire_centres_list,
```

Comment:
This structure seems to have a lot of duplication? I would have thought something like:
`
{
    "zones": [
        {
            "name": "i am a zone",
            "planning_areas": [
                {
                    "name": "i am a planning area",
                    "stations": [
                        {
                            "name": "i am a station"
                        }
                    ]
                }
            ]
        }

    ]
}
` ?